spark-statsd-reporter/src/main/java/org/apache/spark/metrics/sink/statsd/StatsdReporter.java

```diff
@@ -112,66 +88,71 @@ private void reportCounters(SortedMap<String, Counter> counters, DatagramSocket
-                    metricFormatter.buildMetricString(name, "m15_rate", convertRate(timer.getFifteenMinuteRate()), TIMER),
-                    metricFormatter.buildMetricString(name, "mean_rate", convertRate(timer.getMeanRate()), TIMER)
-            );
+            List<String> metrics = new ArrayList<>();
+            addIfMatches(metrics, MetricAttribute.MAX, metricFormatter.buildMetricString(name, MetricAttribute.MAX.toString(), convertDuration(snapshot.getMax()), TIMER));
+            addIfMatches(metrics, MetricAttribute.MEAN, metricFormatter.buildMetricString(name, MetricAttribute.MEAN.toString(), convertDuration(snapshot.getMean()), TIMER));
+            addIfMatches(metrics, MetricAttribute.MIN, metricFormatter.buildMetricString(name, MetricAttribute.MIN.toString(), convertDuration(snapshot.getMin()), TIMER));
+            addIfMatches(metrics, MetricAttribute.STDDEV, metricFormatter.buildMetricString(name, MetricAttribute.STDDEV.toString(), convertDuration(snapshot.getStdDev()), TIMER));
+            addIfMatches(metrics, MetricAttribute.P50, metricFormatter.buildMetricString(name, MetricAttribute.P50.toString(), convertDuration(snapshot.getMedian()), TIMER));
+            addIfMatches(metrics, MetricAttribute.P75, metricFormatter.buildMetricString(name, MetricAttribute.P75.toString(), convertDuration(snapshot.get75thPercentile()), TIMER));
+            addIfMatches(metrics, MetricAttribute.P95, metricFormatter.buildMetricString(name, MetricAttribute.P95.toString(), convertDuration(snapshot.get95thPercentile()), TIMER));
+            addIfMatches(metrics, MetricAttribute.P98, metricFormatter.buildMetricString(name, MetricAttribute.P98.toString(), convertDuration(snapshot.get98thPercentile()), TIMER));
+            addIfMatches(metrics, MetricAttribute.P99, metricFormatter.buildMetricString(name, MetricAttribute.P99.toString(), convertDuration(snapshot.get99thPercentile()), TIMER));
+            addIfMatches(metrics, MetricAttribute.P999, metricFormatter.buildMetricString(name, MetricAttribute.P999.toString(), convertDuration(snapshot.get999thPercentile()), TIMER));
+        	addIfMatches(metrics, MetricAttribute.M1_RATE, metricFormatter.buildMetricString(name, MetricAttribute.M1_RATE.toString(), convertRate(timer.getOneMinuteRate()), TIMER));
+        	addIfMatches(metrics, MetricAttribute.M5_RATE, metricFormatter.buildMetricString(name, MetricAttribute.M5_RATE.toString(), convertRate(timer.getFiveMinuteRate()), TIMER));
+        	addIfMatches(metrics, MetricAttribute.M15_RATE, metricFormatter.buildMetricString(name, MetricAttribute.M15_RATE.toString(), convertRate(timer.getFifteenMinuteRate()), TIMER));
+        	addIfMatches(metrics, MetricAttribute.MEAN_RATE, metricFormatter.buildMetricString(name, MetricAttribute.MEAN_RATE.toString(), convertRate(timer.getMeanRate()), TIMER));
+            send(socket, metrics.toArray(new String[metrics.size()]));
```

Comment:
same re: `metrics.toArray(new String[0])`

Reply:
Changed it accordingly.